web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/svp/SvangerskapspengerTjeneste.java

```diff
@@ -150,27 +186,47 @@ private boolean harSaksbehandletTilrettelegging(Behandling behandling) {
         return aksjonspunkt.isPresent() && aksjonspunkt.get().erUtført();
     }
 
-    private SvpArbeidsforholdDto mapTilretteleggingsinfo(SvpTilretteleggingEntitet svpTilrettelegging, List<Inntektsmelding> inntektsmeldinger, List<SvpTilretteleggingEntitet> opprinneligeTilr,
-                                                         Behandling behandling) {
-        var dto = new SvpArbeidsforholdDto();
-        dto.setTilretteleggingId(svpTilrettelegging.getId());
-        dto.setTilretteleggingBehovFom(svpTilrettelegging.getBehovForTilretteleggingFom());
-        dto.setTilretteleggingDatoer(utledTilretteleggingDatoer(svpTilrettelegging, opprinneligeTilr, behandling));
-        dto.setAvklarteOppholdPerioder(mapAvklartOppholdPeriode(svpTilrettelegging));
+    private SvpArbeidsforholdDto mapArbeidsforholdDto(SvpTilretteleggingEntitet svpTilrettelegging, Behandling behandling) {
+        //Hent informasjon
+        var behandlingId = behandling.getId();
+        var aktørId = behandling.getAktørId();
+        var opprinneligeTilrettelegginger = hentOpprinneligeTilrettlegginger(behandlingId);
+        var iayGrunnlag = iayTjeneste.hentGrunnlag(behandlingId);
```

Comment:
Er det ikke like greit å hente iaygrunnlaget og inntektsmeldinger på toppnivå og sende det ned til SvpArbforhold isf å hente det for hvert tilretteleggin/arbforhold? Det er konstant for en behandling

Reply:
Fikset